test/switch.civet

```diff
@@ -776,6 +777,31 @@ describe "switch", ->
           console.log("empty")}
     """
 
+    testCase """
+      empty array in object
+      ---
+      switch x
+        {children: []}
+          console.log "empty"
+      ---
+      function len<T extends readonly unknown[], N extends number>(arr: T, length: N): arr is T & { length: N } { return arr.length === length }
+      if(typeof x === 'object' && x != null && 'children' in x && Array.isArray(x.children) && len(x.children, 0)) {
+          const {children} = x;
```

Comment:
Oops, I missed this earlier but it should probably compile to `const {children: []}` otherwise it will shadow a possible `children` declared in the outer scope. Same for empty objects as well.

Reply:
Oh, whoops, sorry. My bad adding commits. I hadn't realized the difference but makes sense.